File: 018)JavaMaps/app/src/main/java/com/example/javamaps/adapter/PlaceAdapter.java
package com.example.javamaps.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.javamaps.R;
import com.example.javamaps.databinding.RecyclerRowBinding;
import com.example.javamaps.model.Place;
import com.example.javamaps.view.MapsActivity;

import java.util.List;

public class PlaceAdapter extends RecyclerView.Adapter<PlaceAdapter.PlaceHolder>{
    private Context mContext;
    private List<Place> placeList;


    public PlaceAdapter(Context mContext, List<Place> placeList) {
        this.mContext = mContext;
        this.placeList = placeList;
    }

    @NonNull
    @Override
    public PlaceHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        RecyclerRowBinding binding = DataBindingUtil.inflate(layoutInflater, R.layout.recycler_row, parent, false);
        return new PlaceHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull PlaceHolder holder, int position) {
        Place place = placeList.get(position);
        RecyclerRowBinding binding = holder.binding;
        binding.textViewRV.setText(place.getName());
        binding.cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, MapsActivity.class);
                intent.putExtra("info", "old");
                intent.putExtra("place", place);
                mContext.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        if (placeList == null){
            return 0;
        }else {
            return placeList.size();
        }
    }

    public class PlaceHolder extends RecyclerView.ViewHolder{
        RecyclerRowBinding binding;

        public PlaceHolder(RecyclerRowBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: 058)KotlinMaps/app/src/main/java/com/elyesasimsek/kotlinmaps/adapter/PlaceAdapter.kt
package com.elyesasimsek.kotlinmaps.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.elyesasimsek.kotlinmaps.R
import com.elyesasimsek.kotlinmaps.databinding.RecyclerRowBinding
import com.elyesasimsek.kotlinmaps.model.Place
import com.elyesasimsek.kotlinmaps.view.MapsActivity

class PlaceAdapter (
    private var context: Context,
    private var placeList: List<Place>
): RecyclerView.Adapter<PlaceAdapter.PlaceHolder>(){
    class PlaceHolder(val binding: RecyclerRowBinding): RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlaceHolder {
        val layoutInflater = LayoutInflater.from(context)
        val binding = DataBindingUtil.inflate<RecyclerRowBinding>(layoutInflater,R.layout.recycler_row, parent, false)
        return PlaceHolder(binding)
    }

    override fun getItemCount(): Int {
        if (placeList == null){
            return 0;
        }else{
            return placeList.size
        }
    }

    override fun onBindViewHolder(holder: PlaceHolder, position: Int) {
        val place = placeList[position]
        val binding = holder.binding

        binding.textViewRV.text = place.name
        binding.cardView.setOnClickListener {
            val intent = Intent(context, MapsActivity::class.java)
            intent.putExtra("selectedPlace", place)
            intent.putExtra("info", "old")
            context.startActivity(intent)
        }
    }


}